nav2_behavior_tree/include/nav2_behavior_tree/behavior_tree_engine.hpp

```diff
@@ -44,25 +44,23 @@ class BehaviorTreeEngine
     const std::string & xml_string,
     BT::Blackboard::Ptr blackboard);
 
+  // In order to re-run a Behavior Tree, we must be able to reset all nodes to the initial state
   void haltAllActions(BT::TreeNode * root_node)
   {
+    // this halt signal should propagate through the entire tree.
+    root_node->halt();
+
+    // but, just in case...
     auto visitor = [](BT::TreeNode * node) {
         if (auto action = dynamic_cast<BT::CoroActionNode *>(node)) {
-          action->halt();
+          if( action->status()==BT::NodeStatus::RUNNING) {
```

Comment:
Shouldn't we remove the entire `visitor` declaration? The "just in case" isn't good justification. Also, its casted to `Coro` not your updated `ActionNode`

Reply:
oh, you aright, I need to fix that. Commit coming in minutes...